---
title: "Annotate"
author: "Chuck Roesel"
date: "4/5/2018"
output:
  pdf_document:
    df_print: kable
---

```{r setup, include=T}
rm(list = ls())
knitr::opts_chunk$set(echo = TRUE)
library(tidyr)
library(knitr)
library(dplyr)
library(rjson)
library(stringr)
library(kableExtra)
```

Load concatenated BLAST Output of all four species. Each has a unique pattern for transcript IDs so they can be separated for counts, etc. from the resulting dataframe.

```{r}
spBlast <- read.table("allBlast.tsv", sep="\t", header=F, quote="")
trBlast <- read.table("trBlast.tsv", sep="\t", header=F, quote="")
head(spBlast)
head(trBlast)
```

Assign column names

```{r}
colnames(spBlast) <- c("qseqid","sseqid","slen","qlen","length","pident",
                       "evalue","bitscore","qstart","qend","sstart","send")
colnames(trBlast) <- c("qseqid","sseqid","slen","length","pident","evalue",
                       "bitscore")
head(spBlast)
head(trBlast)
```

Separate pipe-delimited SwissProt IDs

```{r}
spBlast <- separate(spBlast, sseqid, c("giPrefix", "gi", "spPrefix", "sp", "symbol"), sep = "\\|")
head(spBlast)
```
Separate the main SwissProt accession and version suffix

```{r}
trBlast <- separate(trBlast, sseqid, c("spPrefix", "sp", "symbol"), sep = "\\|")
head(trBlast)
```

```{r}
spBlast <- separate(spBlast, sp, c("sp", "version"), sep = "\\.")
head(spBlast)
```
```{r}
spBlast <- separate(spBlast, symbol, c("symbol", "species"), sep = "_")
trBlast <- separate(trBlast, symbol, c("symbol", "species"), sep = "_")
head(spBlast)
head(trBlast)
```
Select the columns I want

```{r}
spBlast <- subset(spBlast, select=c("qseqid", "sp", "symbol","species", "slen", "length", "evalue"))
trBlast <- subset(trBlast, select=c("qseqid", "sp", "symbol","species", "slen", "length", "evalue"))
head(spBlast)
head(trBlast)
```
```{r}
spTrBlast <- rbind(spBlast, trBlast)
tail(spTrBlast)
tail(trBlast)
write.csv(spTrBlast, file="spTrBlast.csv")
```

Calculate coverage

```{r}
spTrBlast$cov <- spTrBlast$length/spTrBlast$slen
head(spTrBlast)
```

Run getKeggUniTable.pl

Load the table of UniProt to KEGG Mappings

```{r}
uniToKegg <- read.table("uniprot.tsv", header=F, sep="\t", 
                        quote="", stringsAsFactors = F)
colnames(uniToKegg) <- c("sp", "kegg")
uniToKegg <- separate(uniToKegg, sp, c("source", "sp"), sep = ":")
head(uniToKegg)
```


Set a coverage threshold, then select the transcript/swissProt pairs with the highest coverage.

```{r}
spTrBlast <- subset(spTrBlast, cov > .5)
spTrBlast <- spTrBlast %>% group_by(qseqid, sp) %>% slice(which.max(cov))
head(spTrBlast)
```
Seprate a prefix to identify each species
```{r}
spTrBlast2 <- separate(spTrBlast, qseqid, c("prefix", "suffix"), sep = 3, remove=F)
head(spTrBlast2)
```
Check the prefix. There should just be one per input species.
```{r}
speciesPrefix <- unique(spTrBlast2$prefix)
speciesPrefix
```
Merge the BLAST table and UniProt to KEGG

```{r}
spTrKeggMerged <- merge(spTrBlast2, uniToKegg)
head(spTrKeggMerged)
```

Create a tsv of the unique KEGG IDs for which detailed entries are needed

```{r}
koToRetrieve <- unique(subset(spTrKeggMerged, select=c("kegg")))
write.table(koToRetrieve, sep="\t", quote=F, row.names =F, col.names = F, file="koToRetrieve.tsv")
```

Run Perl scripts getKeggDetails.pl and parseKo.pl

```{r}
geneToKo <- read.table("geneToKO.tsv", sep="\t", quote="", header=T)
geneToKo <- separate(geneToKo, KO, c("KO", "GeneDesc"), sep = 6)
geneToKo$GeneDesc <- trimws(geneToKo$GeneDesc, which = c("both"))
head(geneToKo)
```

Load the KEGG gene to KEGG Pathway table

```{r}
keggPathways <- read.table("koToPathway.tsv", sep='\t', header=TRUE, quote='')
keggPathways$PathDesc <- paste0('ko',substr(keggPathways$Path, 4,8),' ', keggPathways$PathDesc)
keggPathways <- subset(keggPathways, select=c("kegg", "PathDesc"))
head(keggPathways)
```

Get counts of total genes by pathway

```{r}
koToPathway2 <- merge(geneToKo, keggPathways)
write.csv(koToPathway2, file="koPathway2.csv")
koPathwayTotal <- unique(subset(koToPathway2, select=c("KO", "PathDesc")))
write.csv(koPathwayTotal, file="koPathwayTotal.csv")
byPathTotal <- aggregate(KO ~ PathDesc, data = koPathwayTotal, FUN = NROW)
head(byPathTotal)
```

Get KEGG Pathway Categories using KEGG API

```{r}
pathCategories <- data.frame(PathDesc=character(), Category1=character(), Category=character()) 
data1 <- fromJSON(file="http://www.kegg.jp/kegg-bin/download_htext?htext=br08901.keg&format=json&filedir=")
for (node1 in data1$children){
  for (node2 in node1$children){
    for (node3 in node2$children){
      pathName <- paste0("ko", node3$name)
      pathName <- str_replace(pathName, "  ", " ")
      newRow <- data.frame(PathDesc=pathName,Category1=node1$name, Category=node2$name)
      pathCategories <- rbind(pathCategories,newRow)
    }
  }
}

keggPathways <- merge(keggPathways, pathCategories)
write.csv(pathCategories, file="pathCategories.csv")
head(keggPathways)
```

Get table of unique pathways and categories

```{r}
pathCategory <- unique(subset(keggPathways, select=c("PathDesc", "Category1", "Category")))
write.csv(pathCategory, file="pathCategory")
head(pathCategory)
```

Merge BLAST info with KO and take max coverage

```{r}
uniKeggPathway <- merge(spTrKeggMerged, geneToKo)
head(uniKeggPathway)
```

```{r}
uniKeggPathway <- uniKeggPathway %>% group_by(prefix, KO) %>% slice(which.max(cov))
head(uniKeggPathway)
```

```{r}
uniKeggPathway <- subset(uniKeggPathway, select=-c(kegg, source, suffix))
head(uniKeggPathway)
```

KEGG Ortholog Counts by Prefix

```{r}
byPrefix <- aggregate(KO ~ prefix, data = uniKeggPathway, FUN = NROW)
byPrefix
```

```{r}
genomeCompare <- data.frame()
genomeComparePct <- data.frame()
for (org1 in unique(byPrefix$prefix)){
    org1Genes <- subset(uniKeggPathway, prefix == org1)
  for (org2 in unique(byPrefix$prefix)){
    org2Genes <- subset(uniKeggPathway, prefix == org2)
    org1Unique <- unique(org1Genes$KO)
    org2Unique <- unique(org2Genes$KO)
    commonCount <- length(intersect(org1Unique, org2Unique))
    unionCount <- length(union(org1Unique, org2Unique))
    percentCommon <- (commonCount/unionCount)*100
    genomeCompare[org1, org2] <- commonCount
    genomeComparePct[org1, org2] <- percentCommon
  }
}
kable(genomeCompare, "latex",caption="Shared KEGG Ortholog Counts", row.names = F, booktabs=T) %>% 
  kable_styling("striped", full_width = F)
```
```{r}
kable(genomeComparePct, "latex",caption="Shared KEGG Ortholog Percentages", row.names = F, booktabs=T, digits=1) %>% 
  kable_styling("striped", full_width = F)
```

```{r}
save.image("Annotate.RData")
```

