<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Charles Roesel" />
<meta name="author" content="Steve Vollmer" />

<meta name="date" content="2019-03-07" />

<title>Annotating Transcriptomes with Trans2Kegg</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>

</head>

<body>




<h1 class="title toc-ignore">Annotating Transcriptomes with Trans2Kegg</h1>
<h4 class="author"><em>Charles Roesel</em></h4>
<h4 class="author"><em>Steve Vollmer</em></h4>
<h4 class="date"><em>2019-03-07</em></h4>



<div id="vignette-info" class="section level2">
<h2>Vignette Info</h2>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>Trans2Kegg annotates differential expression (DE) results to KEGG <span class="citation">(Kanehisa et al. 2019, 2017; Kanehisa and Goto 2000)</span> orthologs and pathways requiring only two inputs - a DESeqDataSet <span class="citation">(Love, Huber, and Anders 2014)</span>, and the transcriptome FASTA file to which RNA-Seq reads were aligned. Trans2Kegg extracts all comparisons from the DESeqDataSet and is therefore suitable for single-factor or multi-factor experiments. Two BLAST <span class="citation">(Altschul et al. 1990; “Database Resources of the National Center for Biotechnology Information.” 2016)</span> options are available - BLAST on NCBI servers, or BLAST on AWS using an NCBI BLAST AMI. The AWS interface speeds up the annotation process approximately 14-fold over the NCBI servers. NCBI BLAST AMI requires the user to have an AWS account and launch an EC2 instance using the latest NCBI BLAST image. Two parameters are required from the running instance - the instance ID and the DNS.</p>
<p>Trans2Kegg merges all comparisons from the DESeqDataSet into a single dataframe with an added column to indicate the comparison associated with each DE row. This dataframe is saved as <code>dfAll.csv</code>. The IDs of DE genes (rownames of <code>dfAll.csv</code>) are passed to the <code>annotateAWS</code> function, which extracts IDs of DE genes from the DESeqDataSet, extracts the corresponding sequences from the transcriptome FASTA file, then aligns them to SwissProt <span class="citation">(UniProt Consortium 2018)</span> using the blastSequences function of the BioConductor annotate <span class="citation">(Gentleman 2018)</span> package. Trans2Kegg then queries the KEGG API using the KEGGREST <span class="citation">(Tenenbaum 2018)</span> package to find KEGG ortholog matches for each of the SwissProt BLAST matches, and the results are written to the user-specified output file. We chose KEGG orthologs as the common set of gene identifiers to facilitate cross-species comparisons of DE results. The BLAST and KEGG steps return a list of multiple SwissProt species matches which will generally be associated with one to three KEGG ortholog matches.</p>
<p>To filter these matches, Trans2Kegg first applies e-value and query coverage cutoffs. The default values are e-value less than 1e-10 and query coverage greater than 50%, but the user may optionally pass different cutoff parameters. After applying the e-value and coverage filters, Trans2Kegg then selects the KEGG ortholog with the greatest number of BLAST hits for each transcript, and for each KEGG ortholog the transcript with the greatest number of BLAST hits. Where the number of BLAST hits is equal, for each transcript the KEGG ortholog with the highest average query coverage is selected, and the transcript with the highest average coverage for each ortholog is selected. This reciprocal filtering process eliminates the duplicate DE counting that can occur with simple e-value BLAST cutoffs. The filtered BLAST hits are then merged with the DE data to produce <code>deCovAndCountDesc.csv</code>, a table of 1:1 best matches between transcripts and KEGG orthologs.</p>
<p>To associate orthologs with KEGG pathways, the <code>getPathways</code> function queries the KEGG API using KEGGREST for each of the DE orthologs, producing <code>dfPathsKos.csv</code>. To provide higher-level groupings of pathways like “Immune system” or “Signal transduction” getPathways also queries KEGG for each pathway with DE genes, producing <code>dfPaths.csv</code>. The <code>mergePaths</code> function merges and summarizes DE, ortholog, pathway, and pathway class information to provide up-regulated/down-regulated DE counts by pathway and pathway class, as well as producing a detail table of all merged results.</p>
<p>Aligning transcripts via BLAST is time-consuming, so we allowed for the possibility that R scripts using this package may time-out or experience other network connectivity errors during the annotation process. Each BLAST hit is appended to <code>annot.csv</code> as soon as it is received. If the script times out or errors out and is restarted, it will start BLASTing where it left off. The restart feature also ensures that if you change parameters - for example you start with a very restrictive DE p-value and fold change cutoff, then make it less stringent, the BLAST step only needs to annotate the additional DE genes, not re-annotate the entire list.</p>
</div>
<div id="identify-de-genes" class="section level3">
<h3>Identify DE genes</h3>
<p>The <code>annotateDE</code> function is a helper function that accepts a DESeqDataSet as input, and combines all comparisons into a single dataframe by adding a “Factor” column to indicate the comparison applicable to each row. The output table <code>dfAll.csv</code> includes the ID number associated with the gene, the log<sub>2</sub> fold change, the adjusted p-value, and the comparison.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="co"># Load the Trans2Kegg library</span></a>
<a class="sourceLine" id="cb1-2" title="2"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb1-3" title="3"><span class="kw">library</span>(Trans2Kegg)</a>
<a class="sourceLine" id="cb1-4" title="4"><span class="kw">library</span>(knitr)</a>
<a class="sourceLine" id="cb1-5" title="5"><span class="kw">library</span>(tidyr)</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">annotateDE</span>(ddsAll)</a></code></pre></div>
</div>
<div id="blast-and-get-kegg" class="section level3">
<h3>BLAST and get KEGG</h3>
<p>The annotateTranscript function requires three parameters - 1) a vector of accession numbers from the FASTA file, 2) the path to the FASTA file and 3) the output filename.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1">deAll &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;dfAll.csv&quot;</span>, <span class="dt">row.names=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb3-2" title="2"><span class="co"># Get the IDs (rownames) from deAll.</span></a>
<a class="sourceLine" id="cb3-3" title="3">ids &lt;-<span class="st"> </span><span class="kw">rownames</span>(deAll)</a>
<a class="sourceLine" id="cb3-4" title="4"><span class="co"># Select a subset of rows for testing purposes</span></a>
<a class="sourceLine" id="cb3-5" title="5">ids2 &lt;-<span class="st"> </span><span class="kw">head</span>(ids, <span class="dt">n=</span><span class="dv">10</span>)</a>
<a class="sourceLine" id="cb3-6" title="6"><span class="kw">annotateNCBI</span>(ids2, <span class="st">&quot;aiptasia.fa&quot;</span>)</a></code></pre></div>
</div>
<div id="optional-fast-blast-with-aws" class="section level3">
<h3>Optional: Fast BLAST with AWS</h3>
<p>The annotation process can be sped up dramatically using NCBI BLAST AMI. To setup the AMI, first create an AWS account and login, select the EC2 link (Fig. 2), search for NCBI within the public AMIs (Fig. 3). Check the box to the left of the most recent AMI (as of this writing dated April 11, 2018) and select launch from the Actions drop-down. Select the desired instance type. The type used for the performance tests was m4.2xlarge. Select defaults for other settings, with the exception of security group. Click “Add Rule” and select HTTP (Fig. 4). After launching the instance copy and paste the instance ID and DNS from the EC2 console (Fig. 1) to the instance and dns variables passed to annotateAws. The annotateAws function requires three parameters - IDs to annotate (ids2), FASTA file of transcripts (fastaFile), instance ID, and DNS ID. Optionally pass the threads parameter to change from the default of four threads.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1">deAll &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;dfAll.csv&quot;</span>, <span class="dt">row.names=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb4-2" title="2"><span class="co"># Get the IDs (rownames) from deAll.</span></a>
<a class="sourceLine" id="cb4-3" title="3">ids &lt;-<span class="st"> </span><span class="kw">rownames</span>(deAll)</a>
<a class="sourceLine" id="cb4-4" title="4"><span class="co"># Select a subset of rows for testing purposes</span></a>
<a class="sourceLine" id="cb4-5" title="5">ids2 &lt;-<span class="st"> </span><span class="kw">head</span>(ids, <span class="dt">n=</span><span class="dv">200</span>)</a>
<a class="sourceLine" id="cb4-6" title="6"><span class="co"># </span><span class="al">NOTE</span><span class="co">: instance and DNS show below are examples only. </span></a>
<a class="sourceLine" id="cb4-7" title="7"><span class="co"># Users must launch their own NCBI BLAST AMI</span></a>
<a class="sourceLine" id="cb4-8" title="8">instance &lt;-<span class="st"> &#39;i-07da948c2d85b7388&#39;</span></a>
<a class="sourceLine" id="cb4-9" title="9">dns &lt;-<span class="st"> &#39;ec2-54-175-9-203.compute-1.amazonaws.com&#39;</span></a>
<a class="sourceLine" id="cb4-10" title="10"><span class="kw">annotateAWS</span>(ids2, <span class="st">&quot;aiptasia.fa&quot;</span>, <span class="dt">instance=</span>instance, <span class="dt">dns=</span>dns, <span class="dt">threads=</span><span class="dv">2</span>)</a></code></pre></div>
</div>
<div id="merge-annotations" class="section level3">
<h3>Merge Annotations</h3>
<p>Merge the BLAST and KEGG annotations with DE results, and filter to obtain 1:1 associations between transcript IDs and KEGG orthologs (Table 3).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw">mergeAnnotations</span>()</a></code></pre></div>
</div>
<div id="get-kegg-pathway-information" class="section level3">
<h3>Get KEGG pathway information</h3>
<p>Query the KEGG API to obtain pathway information, and produce a table of pathway to ortholog mappings (Table 4), and pathway ID to pathway detail mappings (Table 5).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="kw">getPathways</span>()</a></code></pre></div>
</div>
<div id="merge-and-summarize-all-annotations" class="section level3">
<h3>Merge and summarize all annotations</h3>
<p>Merge the DE, KEGG ortholog, and pathway and provide summary counts.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"><span class="kw">mergePaths</span>() </a></code></pre></div>
</div>
<div id="sample-output" class="section level3">
<h3>Sample Output</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">deAll &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;dfAll.csv&quot;</span>, <span class="dt">row.names=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-2" title="2"><span class="kw">kable</span>(<span class="kw">head</span>(deAll), <span class="dt">caption =</span> <span class="st">&quot;Sample of deAll.csv, the output of function </span></a>
<a class="sourceLine" id="cb8-3" title="3"><span class="st">    AnnotateDE, which combines all comparisons from a DESeqDataSet</span></a>
<a class="sourceLine" id="cb8-4" title="4"><span class="st">    into a single file to facilitate annotation.&quot;</span>, <span class="dt">booktabs=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<table>
<caption>Sample of deAll.csv, the output of function AnnotateDE, which combines all comparisons from a DESeqDataSet into a single file to facilitate annotation.</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">log2FoldChange</th>
<th align="right">padj</th>
<th align="left">Factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KXJ04205.1</td>
<td align="right">-2.114027</td>
<td align="right">0.0008937</td>
<td align="left">Vibrio_Vibrio_vs_NoVibrio</td>
</tr>
<tr class="even">
<td>KXJ04307.1</td>
<td align="right">2.246925</td>
<td align="right">0.0000055</td>
<td align="left">Vibrio_Vibrio_vs_NoVibrio</td>
</tr>
<tr class="odd">
<td>KXJ04315.1</td>
<td align="right">1.156684</td>
<td align="right">0.0010343</td>
<td align="left">Vibrio_Vibrio_vs_NoVibrio</td>
</tr>
<tr class="even">
<td>KXJ04345.1</td>
<td align="right">5.113863</td>
<td align="right">0.0002507</td>
<td align="left">Vibrio_Vibrio_vs_NoVibrio</td>
</tr>
<tr class="odd">
<td>KXJ04348.1</td>
<td align="right">1.412197</td>
<td align="right">0.0002342</td>
<td align="left">Vibrio_Vibrio_vs_NoVibrio</td>
</tr>
<tr class="even">
<td>KXJ04356.1</td>
<td align="right">2.223019</td>
<td align="right">0.0000045</td>
<td align="left">Vibrio_Vibrio_vs_NoVibrio</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">outFile &lt;-<span class="st"> &quot;annot.csv&quot;</span></a>
<a class="sourceLine" id="cb9-2" title="2">annot &lt;-<span class="st"> </span><span class="kw">read.csv</span>(outFile, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb9-3" title="3"><span class="kw">kable</span>(<span class="kw">head</span>(annot, <span class="dt">n=</span> 20L), </a>
<a class="sourceLine" id="cb9-4" title="4">    <span class="dt">caption=</span><span class="st">&quot;Sample annot.csv output of annotateTranscript and annotateAws </span></a>
<a class="sourceLine" id="cb9-5" title="5"><span class="st">    functions.&quot;</span>, <span class="dt">booktab=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<table>
<caption>Sample annot.csv output of annotateTranscript and annotateAws functions.</caption>
<thead>
<tr class="header">
<th align="left">uniprot</th>
<th align="left">kegg</th>
<th align="left">ko</th>
<th align="left">desc</th>
<th align="left">Iteration_query.def</th>
<th align="right">Iteration_query.len</th>
<th align="right">Hit_len</th>
<th align="right">Hsp_evalue</th>
<th align="right">Hsp_identity</th>
<th align="right">Hsp_gaps</th>
<th align="right">Hsp_align.len</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Q9CZV8</td>
<td align="left">mmu:72194</td>
<td align="left">K10268</td>
<td align="left">F-box and leucine-rich repeat protein 2/20</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">436</td>
<td align="right">0.0e+00</td>
<td align="right">86</td>
<td align="right">17</td>
<td align="right">327</td>
</tr>
<tr class="even">
<td align="left">Q96IG2</td>
<td align="left">hsa:84961</td>
<td align="left">K10268</td>
<td align="left">F-box and leucine-rich repeat protein 2/20</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">436</td>
<td align="right">0.0e+00</td>
<td align="right">85</td>
<td align="right">17</td>
<td align="right">327</td>
</tr>
<tr class="odd">
<td align="left">A1A5X2</td>
<td align="left">dre:569430</td>
<td align="left">K10273</td>
<td align="left">F-box and leucine-rich repeat protein 7</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">489</td>
<td align="right">0.0e+00</td>
<td align="right">94</td>
<td align="right">50</td>
<td align="right">421</td>
</tr>
<tr class="even">
<td align="left">A6H779</td>
<td align="left">bta:616212</td>
<td align="left">K10268</td>
<td align="left">F-box and leucine-rich repeat protein 2/20</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">423</td>
<td align="right">0.0e+00</td>
<td align="right">93</td>
<td align="right">52</td>
<td align="right">373</td>
</tr>
<tr class="odd">
<td align="left">Q8BH16</td>
<td align="left">mmu:72179</td>
<td align="left">K10268</td>
<td align="left">F-box and leucine-rich repeat protein 2/20</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">423</td>
<td align="right">0.0e+00</td>
<td align="right">91</td>
<td align="right">21</td>
<td align="right">329</td>
</tr>
<tr class="even">
<td align="left">Q9QZH7</td>
<td align="left">rno:64039</td>
<td align="left">K10268</td>
<td align="left">F-box and leucine-rich repeat protein 2/20</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">276</td>
<td align="right">0.0e+00</td>
<td align="right">71</td>
<td align="right">12</td>
<td align="right">244</td>
</tr>
<tr class="odd">
<td align="left">Q5BJ29</td>
<td align="left">mmu:448987</td>
<td align="left">K10273</td>
<td align="left">F-box and leucine-rich repeat protein 7</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">491</td>
<td align="right">0.0e+00</td>
<td align="right">103</td>
<td align="right">63</td>
<td align="right">431</td>
</tr>
<tr class="even">
<td align="left">Q9UJT9</td>
<td align="left">hsa:23194</td>
<td align="left">K10273</td>
<td align="left">F-box and leucine-rich repeat protein 7</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">491</td>
<td align="right">0.0e+00</td>
<td align="right">103</td>
<td align="right">63</td>
<td align="right">431</td>
</tr>
<tr class="odd">
<td align="left">Q9UKC9</td>
<td align="left">hsa:25827</td>
<td align="left">K10268</td>
<td align="left">F-box and leucine-rich repeat protein 2/20</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">423</td>
<td align="right">0.0e+00</td>
<td align="right">89</td>
<td align="right">21</td>
<td align="right">329</td>
</tr>
<tr class="even">
<td align="left">Q5R3Z8</td>
<td align="left">pon:100174084</td>
<td align="left">K10268</td>
<td align="left">F-box and leucine-rich repeat protein 2/20</td>
<td align="left">KXJ29331.1</td>
<td align="right">1782</td>
<td align="right">423</td>
<td align="right">0.0e+00</td>
<td align="right">89</td>
<td align="right">21</td>
<td align="right">329</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">KXJ30207.1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">Q8C7R7</td>
<td align="left">mmu:320995</td>
<td align="left">K19521</td>
<td align="left">regulatory factor X 6</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">927</td>
<td align="right">0.0e+00</td>
<td align="right">83</td>
<td align="right">1</td>
<td align="right">133</td>
</tr>
<tr class="odd">
<td align="left">D2HNW6</td>
<td align="left">aml:100476328</td>
<td align="left">K19521</td>
<td align="left">regulatory factor X 6</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">928</td>
<td align="right">0.0e+00</td>
<td align="right">83</td>
<td align="right">1</td>
<td align="right">133</td>
</tr>
<tr class="even">
<td align="left">Q8HWS3</td>
<td align="left">hsa:222546</td>
<td align="left">K19521</td>
<td align="left">regulatory factor X 6</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">928</td>
<td align="right">0.0e+00</td>
<td align="right">83</td>
<td align="right">1</td>
<td align="right">133</td>
</tr>
<tr class="odd">
<td align="left">Q7TNK1</td>
<td align="left">mmu:71137</td>
<td align="left">K09174</td>
<td align="left">regulatory factor X 4</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">735</td>
<td align="right">0.0e+00</td>
<td align="right">66</td>
<td align="right">0</td>
<td align="right">121</td>
</tr>
<tr class="even">
<td align="left">A2BGA0</td>
<td align="left">dre:403016</td>
<td align="left">K09174</td>
<td align="left">regulatory factor X 4</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">735</td>
<td align="right">0.0e+00</td>
<td align="right">65</td>
<td align="right">0</td>
<td align="right">121</td>
</tr>
<tr class="odd">
<td align="left">Q33E94</td>
<td align="left">hsa:5992</td>
<td align="left">K09174</td>
<td align="left">regulatory factor X 4</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">735</td>
<td align="right">0.0e+00</td>
<td align="right">62</td>
<td align="right">0</td>
<td align="right">112</td>
</tr>
<tr class="even">
<td align="left">Q0V9K5</td>
<td align="left">xtr:779608</td>
<td align="left">K09173</td>
<td align="left">regulatory factor X 1/2/3</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">749</td>
<td align="right">6.0e-07</td>
<td align="right">34</td>
<td align="right">28</td>
<td align="right">160</td>
</tr>
<tr class="odd">
<td align="left">B1WAV2</td>
<td align="left">xtr:100144734</td>
<td align="left">K09173</td>
<td align="left">regulatory factor X 1/2/3</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">694</td>
<td align="right">2.3e-06</td>
<td align="right">33</td>
<td align="right">23</td>
<td align="right">126</td>
</tr>
<tr class="even">
<td align="left">P48381</td>
<td align="left">mmu:19726</td>
<td align="left">K09173</td>
<td align="left">regulatory factor X 1/2/3</td>
<td align="left">KXJ30097.1</td>
<td align="right">2593</td>
<td align="right">749</td>
<td align="right">3.1e-06</td>
<td align="right">34</td>
<td align="right">28</td>
<td align="right">160</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">dfCovCount &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;cvCnt.csv&quot;</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb10-2" title="2">dfCovCount<span class="op">$</span>Factor &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, dfCovCount<span class="op">$</span>Factor)</a>
<a class="sourceLine" id="cb10-3" title="3"><span class="kw">kable</span>(<span class="kw">head</span>(dfCovCount, <span class="dt">n=</span>20L), </a>
<a class="sourceLine" id="cb10-4" title="4">    <span class="dt">caption=</span><span class="st">&quot;Sample cvCnt.csv output of mergePathways function.</span></a>
<a class="sourceLine" id="cb10-5" title="5"><span class="st">    This table summarizes the annotation results. qCov is the mean query</span></a>
<a class="sourceLine" id="cb10-6" title="6"><span class="st">    coverage for the species BLAST hits matching the KEGG ortholog.</span></a>
<a class="sourceLine" id="cb10-7" title="7"><span class="st">    n is the number of species BLAST hits matching the KEGG ortholog.&quot;</span>, </a>
<a class="sourceLine" id="cb10-8" title="8">    <span class="dt">booktab=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<table>
<caption>Sample cvCnt.csv output of mergePathways function. This table summarizes the annotation results. qCov is the mean query coverage for the species BLAST hits matching the KEGG ortholog. n is the number of species BLAST hits matching the KEGG ortholog.</caption>
<thead>
<tr class="header">
<th align="left">Iteration_query.def</th>
<th align="right">log2FoldChange</th>
<th align="right">padj</th>
<th align="left">Factor</th>
<th align="left">ko</th>
<th align="right">qCov</th>
<th align="right">n</th>
<th align="left">desc</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">KXJ04205.1</td>
<td align="right">-2.114027</td>
<td align="right">0.0008937</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K00069</td>
<td align="right">0.6289201</td>
<td align="right">4</td>
<td align="left">15-hydroxyprostaglandin dehydrogenase (NAD) [EC:1.1.1.141]</td>
</tr>
<tr class="even">
<td align="left">KXJ04879.1</td>
<td align="right">-1.068737</td>
<td align="right">0.0088494</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K01437</td>
<td align="right">0.9388352</td>
<td align="right">6</td>
<td align="left">aspartoacylase [EC:3.5.1.15]</td>
</tr>
<tr class="odd">
<td align="left">KXJ05538.1</td>
<td align="right">1.685325</td>
<td align="right">0.0066262</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K04437</td>
<td align="right">0.5845974</td>
<td align="right">3</td>
<td align="left">filamin</td>
</tr>
<tr class="even">
<td align="left">KXJ05558.1</td>
<td align="right">-1.060388</td>
<td align="right">0.0013819</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K09863</td>
<td align="right">0.8152091</td>
<td align="right">5</td>
<td align="left">aquaporin-0</td>
</tr>
<tr class="odd">
<td align="left">KXJ05589.1</td>
<td align="right">1.053273</td>
<td align="right">0.0046784</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K20865</td>
<td align="right">0.5155514</td>
<td align="right">1</td>
<td align="left">NACHT, LRR and PYD domains-containing protein 12</td>
</tr>
<tr class="even">
<td align="left">KXJ05757.1</td>
<td align="right">-1.910618</td>
<td align="right">0.0009740</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K00546</td>
<td align="right">0.8767264</td>
<td align="right">8</td>
<td align="left">histamine N-methyltransferase [EC:2.1.1.8]</td>
</tr>
<tr class="odd">
<td align="left">KXJ05958.1</td>
<td align="right">2.695624</td>
<td align="right">0.0000066</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K15789</td>
<td align="right">0.7560322</td>
<td align="right">3</td>
<td align="left">threonine 3-dehydrogenase [EC:1.1.1.103]</td>
</tr>
<tr class="even">
<td align="left">KXJ06049.1</td>
<td align="right">1.224146</td>
<td align="right">0.0000081</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K20347</td>
<td align="right">0.8818797</td>
<td align="right">5</td>
<td align="left">p24 family protein beta-1</td>
</tr>
<tr class="odd">
<td align="left">KXJ06110.1</td>
<td align="right">1.688719</td>
<td align="right">0.0002665</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K12649</td>
<td align="right">0.5530973</td>
<td align="right">2</td>
<td align="left">ATP-dependent RNA helicase DHX58 [EC:3.6.4.13]</td>
</tr>
<tr class="even">
<td align="left">KXJ06389.1</td>
<td align="right">1.787778</td>
<td align="right">0.0006306</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K03173</td>
<td align="right">0.9131737</td>
<td align="right">2</td>
<td align="left">TNF receptor-associated factor 2 [EC:2.3.2.27]</td>
</tr>
<tr class="odd">
<td align="left">KXJ06413.1</td>
<td align="right">-1.016276</td>
<td align="right">0.0026431</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K18622</td>
<td align="right">0.6512346</td>
<td align="right">1</td>
<td align="left">adducin</td>
</tr>
<tr class="even">
<td align="left">KXJ06963.1</td>
<td align="right">1.473685</td>
<td align="right">0.0036180</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K06772</td>
<td align="right">0.6272189</td>
<td align="right">3</td>
<td align="left">limbic system-associated membrane protein</td>
</tr>
<tr class="odd">
<td align="left">KXJ07019.1</td>
<td align="right">1.930533</td>
<td align="right">0.0000137</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K06185</td>
<td align="right">0.8497635</td>
<td align="right">8</td>
<td align="left">ATP-binding cassette, subfamily F, member 2</td>
</tr>
<tr class="even">
<td align="left">KXJ07281.1</td>
<td align="right">-1.038218</td>
<td align="right">0.0002820</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K18172</td>
<td align="right">0.7220680</td>
<td align="right">7</td>
<td align="left">COX assembly mitochondrial protein 2</td>
</tr>
<tr class="odd">
<td align="left">KXJ07483.1</td>
<td align="right">1.147990</td>
<td align="right">0.0000013</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K06174</td>
<td align="right">0.9851053</td>
<td align="right">6</td>
<td align="left">ATP-binding cassette, sub-family E, member 1</td>
</tr>
<tr class="even">
<td align="left">KXJ07538.1</td>
<td align="right">5.968846</td>
<td align="right">0.0018749</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K20876</td>
<td align="right">0.6572848</td>
<td align="right">4</td>
<td align="left">NIMA (never in mitosis gene a)-related kinase 7 [EC:2.7.11.1]</td>
</tr>
<tr class="odd">
<td align="left">KXJ07556.1</td>
<td align="right">-1.197735</td>
<td align="right">0.0016098</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K02183</td>
<td align="right">0.7718121</td>
<td align="right">2</td>
<td align="left">calmodulin</td>
</tr>
<tr class="even">
<td align="left">KXJ07696.1</td>
<td align="right">-2.432621</td>
<td align="right">0.0000096</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K01493</td>
<td align="right">0.5393258</td>
<td align="right">3</td>
<td align="left">dCMP deaminase [EC:3.5.4.12]</td>
</tr>
<tr class="odd">
<td align="left">KXJ07902.1</td>
<td align="right">1.439614</td>
<td align="right">0.0001321</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K14355</td>
<td align="right">0.5487251</td>
<td align="right">4</td>
<td align="left">solute carrier organic anion transporter family, member 4C</td>
</tr>
<tr class="even">
<td align="left">KXJ07925.1</td>
<td align="right">1.074620</td>
<td align="right">0.0003135</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">K18032</td>
<td align="right">0.7107232</td>
<td align="right">1</td>
<td align="left">receptor-type tyrosine-protein phosphatase A [EC:3.1.3.48]</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">dfPathsKos &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;pthKo.csv&quot;</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb11-2" title="2"><span class="kw">kable</span>(<span class="kw">head</span>(dfPathsKos, <span class="dt">n=</span>20L), </a>
<a class="sourceLine" id="cb11-3" title="3">    <span class="dt">caption=</span><span class="st">&quot;Sample pthKo.csv output of getPathways function.&quot;</span>, </a>
<a class="sourceLine" id="cb11-4" title="4">    <span class="dt">booktab=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<table>
<caption>Sample pthKo.csv output of getPathways function.</caption>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">ko</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K06256</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K05704</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K06106</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K05083</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K04364</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K02833</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K07827</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K07828</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K07829</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K07830</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K07831</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K04365</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K04366</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K08845</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K04368</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K04369</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K04371</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K04392</td>
</tr>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">K16848</td>
</tr>
<tr class="even">
<td align="left">ko05205</td>
<td align="left">K04393</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">dfPaths &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;path.csv&quot;</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb12-2" title="2"><span class="kw">kable</span>(<span class="kw">head</span>(dfPaths, <span class="dt">n=</span>20L), </a>
<a class="sourceLine" id="cb12-3" title="3">    <span class="dt">caption=</span><span class="st">&quot;Sample path.csv output of getPathways function.&quot;</span>, </a>
<a class="sourceLine" id="cb12-4" title="4">    <span class="dt">booktab=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<table>
<caption>Sample path.csv output of getPathways function.</caption>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">class</th>
<th align="left">path</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ko05205</td>
<td align="left">Human Diseases; Cancers: Overview</td>
<td align="left">Proteoglycans in cancer</td>
</tr>
<tr class="even">
<td align="left">map05205</td>
<td align="left">Human Diseases; Cancers: Overview</td>
<td align="left">Proteoglycans in cancer</td>
</tr>
<tr class="odd">
<td align="left">ko05132</td>
<td align="left">Human Diseases; Infectious diseases: Bacterial</td>
<td align="left">Salmonella infection</td>
</tr>
<tr class="even">
<td align="left">map05132</td>
<td align="left">Human Diseases; Infectious diseases: Bacterial</td>
<td align="left">Salmonella infection</td>
</tr>
<tr class="odd">
<td align="left">ko04510</td>
<td align="left">Cellular Processes; Cellular community - eukaryotes</td>
<td align="left">Focal adhesion</td>
</tr>
<tr class="even">
<td align="left">map04510</td>
<td align="left">Cellular Processes; Cellular community - eukaryotes</td>
<td align="left">Focal adhesion</td>
</tr>
<tr class="odd">
<td align="left">ko04010</td>
<td align="left">Environmental Information Processing; Signal transduction</td>
<td align="left">MAPK signaling pathway</td>
</tr>
<tr class="even">
<td align="left">map04010</td>
<td align="left">Environmental Information Processing; Signal transduction</td>
<td align="left">MAPK signaling pathway</td>
</tr>
<tr class="odd">
<td align="left">ko04610</td>
<td align="left">Organismal Systems; Immune system</td>
<td align="left">Complement and coagulation cascades</td>
</tr>
<tr class="even">
<td align="left">map04610</td>
<td align="left">Organismal Systems; Immune system</td>
<td align="left">Complement and coagulation cascades</td>
</tr>
<tr class="odd">
<td align="left">ko05203</td>
<td align="left">Human Diseases; Cancers: Overview</td>
<td align="left">Viral carcinogenesis</td>
</tr>
<tr class="even">
<td align="left">map05203</td>
<td align="left">Human Diseases; Cancers: Overview</td>
<td align="left">Viral carcinogenesis</td>
</tr>
<tr class="odd">
<td align="left">ko04512</td>
<td align="left">Environmental Information Processing; Signaling molecules and interaction</td>
<td align="left">ECM-receptor interaction</td>
</tr>
<tr class="even">
<td align="left">map04512</td>
<td align="left">Environmental Information Processing; Signaling molecules and interaction</td>
<td align="left">ECM-receptor interaction</td>
</tr>
<tr class="odd">
<td align="left">ko04972</td>
<td align="left">Organismal Systems; Digestive system</td>
<td align="left">Pancreatic secretion</td>
</tr>
<tr class="even">
<td align="left">map04972</td>
<td align="left">Organismal Systems; Digestive system</td>
<td align="left">Pancreatic secretion</td>
</tr>
<tr class="odd">
<td align="left">ko04970</td>
<td align="left">Organismal Systems; Digestive system</td>
<td align="left">Salivary secretion</td>
</tr>
<tr class="even">
<td align="left">map04970</td>
<td align="left">Organismal Systems; Digestive system</td>
<td align="left">Salivary secretion</td>
</tr>
<tr class="odd">
<td align="left">ko00760</td>
<td align="left">Metabolism; Metabolism of cofactors and vitamins</td>
<td align="left">Nicotinate and nicotinamide metabolism</td>
</tr>
<tr class="even">
<td align="left">map00760</td>
<td align="left">Metabolism; Metabolism of cofactors and vitamins</td>
<td align="left">Nicotinate and nicotinamide metabolism</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1">countByPath &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;countByPath.csv&quot;</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb13-2" title="2"><span class="co"># Replace _ with spaces in Factor to allow word-wrap within the column.</span></a>
<a class="sourceLine" id="cb13-3" title="3">countByPath<span class="op">$</span>Factor &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, countByPath<span class="op">$</span>Factor)</a>
<a class="sourceLine" id="cb13-4" title="4"><span class="co"># Subset to include only &quot;Organismal Systems&quot; category</span></a>
<a class="sourceLine" id="cb13-5" title="5">countByPathOrg &lt;-<span class="st"> </span><span class="kw">subset</span>(countByPath, category <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb13-6" title="6"><span class="st">    </span><span class="kw">c</span>(<span class="st">&quot;Organismal Systems&quot;</span>))</a>
<a class="sourceLine" id="cb13-7" title="7">countByPathOrg &lt;-<span class="st"> </span><span class="kw">arrange</span>(countByPathOrg, class, path, direction)</a>
<a class="sourceLine" id="cb13-8" title="8"><span class="kw">kable</span>(<span class="kw">head</span>(countByPathOrg, <span class="dt">n=</span>15L), </a>
<a class="sourceLine" id="cb13-9" title="9">    <span class="dt">caption=</span><span class="st">&quot;Sample countByPath.csv output of mergePathways function.&quot;</span>, </a>
<a class="sourceLine" id="cb13-10" title="10">    <span class="dt">booktab=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<table>
<caption>Sample countByPath.csv output of mergePathways function.</caption>
<thead>
<tr class="header">
<th align="left">Factor</th>
<th align="left">category</th>
<th align="left">class</th>
<th align="left">path</th>
<th align="left">direction</th>
<th align="right">ko</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Aging</td>
<td align="left">Longevity regulating pathway - multiple species</td>
<td align="left">Down</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Aging</td>
<td align="left">Longevity regulating pathway - worm</td>
<td align="left">Down</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Circulatory system</td>
<td align="left">Adrenergic signaling in cardiomyocytes</td>
<td align="left">Down</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Circulatory system</td>
<td align="left">Adrenergic signaling in cardiomyocytes</td>
<td align="left">Up</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Circulatory system</td>
<td align="left">Vascular smooth muscle contraction</td>
<td align="left">Down</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Circulatory system</td>
<td align="left">Vascular smooth muscle contraction</td>
<td align="left">Up</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Development</td>
<td align="left">Axon guidance</td>
<td align="left">Down</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Development</td>
<td align="left">Axon guidance</td>
<td align="left">Up</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Development</td>
<td align="left">Dorso-ventral axis formation</td>
<td align="left">Up</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Development</td>
<td align="left">Osteoclast differentiation</td>
<td align="left">Up</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Digestive system</td>
<td align="left">Bile secretion</td>
<td align="left">Down</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Digestive system</td>
<td align="left">Cholesterol metabolism</td>
<td align="left">Up</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Digestive system</td>
<td align="left">Fat digestion and absorption</td>
<td align="left">Down</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Digestive system</td>
<td align="left">Fat digestion and absorption</td>
<td align="left">Up</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Digestive system</td>
<td align="left">Gastric acid secretion</td>
<td align="left">Down</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="co"># Read in countByClass.csv</span></a>
<a class="sourceLine" id="cb14-2" title="2">countByClass &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;countByClass.csv&quot;</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb14-3" title="3"><span class="co"># Replace _ with spaces in Factor to allow word-wrap within the column.</span></a>
<a class="sourceLine" id="cb14-4" title="4">countByClass<span class="op">$</span>Factor &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, countByClass<span class="op">$</span>Factor)</a>
<a class="sourceLine" id="cb14-5" title="5"><span class="co"># Subset to include only &quot;Organismal Systems&quot; category</span></a>
<a class="sourceLine" id="cb14-6" title="6">countByClassOrg &lt;-<span class="st"> </span><span class="kw">subset</span>(countByClass, category <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-7" title="7"><span class="st">    </span><span class="kw">c</span>(<span class="st">&quot;Organismal Systems&quot;</span>))</a>
<a class="sourceLine" id="cb14-8" title="8">countByClassOrg &lt;-<span class="st"> </span><span class="kw">arrange</span>(countByClassOrg, class, direction)</a>
<a class="sourceLine" id="cb14-9" title="9"><span class="kw">kable</span>(<span class="kw">head</span>(countByClassOrg, <span class="dt">n=</span>20L), </a>
<a class="sourceLine" id="cb14-10" title="10">    <span class="dt">caption=</span><span class="st">&quot;Sample countByClass.csv output of mergePathways function.&quot;</span>, </a>
<a class="sourceLine" id="cb14-11" title="11">    <span class="dt">booktab=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<table>
<caption>Sample countByClass.csv output of mergePathways function.</caption>
<thead>
<tr class="header">
<th align="left">Factor</th>
<th align="left">category</th>
<th align="left">class</th>
<th align="left">direction</th>
<th align="right">ko</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Aging</td>
<td align="left">Down</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Circulatory system</td>
<td align="left">Down</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Circulatory system</td>
<td align="left">Up</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Development</td>
<td align="left">Down</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Development</td>
<td align="left">Up</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Digestive system</td>
<td align="left">Down</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Digestive system</td>
<td align="left">Up</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Endocrine system</td>
<td align="left">Down</td>
<td align="right">21</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Endocrine system</td>
<td align="left">Up</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Environmental adaptation</td>
<td align="left">Down</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Environmental adaptation</td>
<td align="left">Up</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Excretory system</td>
<td align="left">Down</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Excretory system</td>
<td align="left">Up</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Down</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Up</td>
<td align="right">33</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Nervous system</td>
<td align="left">Down</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Nervous system</td>
<td align="left">Up</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Sensory system</td>
<td align="left">Down</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="left">Organismal Systems</td>
<td align="left">Sensory system</td>
<td align="left">Up</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1"><span class="co"># Read in dePathsDetails.csv</span></a>
<a class="sourceLine" id="cb15-2" title="2">deDetails &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;dePathsDetails.csv&quot;</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb15-3" title="3"><span class="co"># Replace _ with spaces in Factor to allow word-wrap within the column.</span></a>
<a class="sourceLine" id="cb15-4" title="4">deDetails<span class="op">$</span>Factor &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, deDetails<span class="op">$</span>Factor)</a>
<a class="sourceLine" id="cb15-5" title="5"><span class="co"># Subset genes with Immune system pathways</span></a>
<a class="sourceLine" id="cb15-6" title="6">deDetailsImmune &lt;-<span class="st"> </span><span class="kw">subset</span>(deDetails, class <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot; Immune system&quot;</span>))</a>
<a class="sourceLine" id="cb15-7" title="7"><span class="kw">kable</span>(<span class="kw">head</span>(deDetailsImmune, <span class="dt">n=</span>10L), </a>
<a class="sourceLine" id="cb15-8" title="8">    <span class="dt">caption=</span><span class="st">&quot;Sample dePathsDetails.csv output of mergePathways function, </span></a>
<a class="sourceLine" id="cb15-9" title="9"><span class="st">    subset to show the first ten genes within Immune system.</span></a>
<a class="sourceLine" id="cb15-10" title="10"><span class="st">    The combinations of DE and KEGG information in dePathsDetails.csv </span></a>
<a class="sourceLine" id="cb15-11" title="11"><span class="st">    facilitates subsetting and summarizing DE results&quot;</span>, </a>
<a class="sourceLine" id="cb15-12" title="12">    <span class="dt">booktab=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<table>
<caption>Sample dePathsDetails.csv output of mergePathways function, subset to show the first ten genes within Immune system. The combinations of DE and KEGG information in dePathsDetails.csv facilitates subsetting and summarizing DE results</caption>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">ko</th>
<th align="left">Iteration_query.def</th>
<th align="right">log2FoldChange</th>
<th align="right">padj</th>
<th align="left">Factor</th>
<th align="right">qCov</th>
<th align="right">n</th>
<th align="left">desc</th>
<th align="left">category</th>
<th align="left">class</th>
<th align="left">path</th>
<th align="left">direction</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ko04062</td>
<td align="left">K04371</td>
<td align="left">KXJ11891.1</td>
<td align="right">2.366390</td>
<td align="right">0.0000062</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.8662747</td>
<td align="right">3</td>
<td align="left">mitogen-activated protein kinase 1/3 [EC:2.7.11.24]</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Chemokine signaling pathway</td>
<td align="left">Up</td>
</tr>
<tr class="even">
<td align="left">ko04062</td>
<td align="left">K08011</td>
<td align="left">KXJ14574.1</td>
<td align="right">2.307743</td>
<td align="right">0.0000145</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.6604134</td>
<td align="right">3</td>
<td align="left">neutrophil cytosolic factor 1</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Chemokine signaling pathway</td>
<td align="left">Up</td>
</tr>
<tr class="odd">
<td align="left">ko04062</td>
<td align="left">K06068</td>
<td align="left">KXJ15884.1</td>
<td align="right">1.740334</td>
<td align="right">0.0000051</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.9232713</td>
<td align="right">5</td>
<td align="left">novel protein kinase C delta type [EC:2.7.11.13]</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Chemokine signaling pathway</td>
<td align="left">Up</td>
</tr>
<tr class="even">
<td align="left">ko04610</td>
<td align="left">K04003</td>
<td align="left">KXJ11932.1</td>
<td align="right">4.093962</td>
<td align="right">0.0005606</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.5793651</td>
<td align="right">1</td>
<td align="left">complement component 4 binding protein, beta</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Complement and coagulation cascades</td>
<td align="left">Up</td>
</tr>
<tr class="odd">
<td align="left">ko04610</td>
<td align="left">K03909</td>
<td align="left">KXJ11082.1</td>
<td align="right">2.028126</td>
<td align="right">0.0003002</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.6377047</td>
<td align="right">3</td>
<td align="left">tissue factor pathway inhibitor</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Complement and coagulation cascades</td>
<td align="left">Up</td>
</tr>
<tr class="even">
<td align="left">ko04610</td>
<td align="left">K03990</td>
<td align="left">KXJ11955.1</td>
<td align="right">2.438592</td>
<td align="right">0.0000000</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.6523754</td>
<td align="right">4</td>
<td align="left">complement component 3</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Complement and coagulation cascades</td>
<td align="left">Up</td>
</tr>
<tr class="odd">
<td align="left">ko04610</td>
<td align="left">K03910</td>
<td align="left">KXJ15600.1</td>
<td align="right">1.858965</td>
<td align="right">0.0000000</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.6133115</td>
<td align="right">2</td>
<td align="left">alpha-2-macroglobulin</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Complement and coagulation cascades</td>
<td align="left">Up</td>
</tr>
<tr class="even">
<td align="left">ko04611</td>
<td align="left">K04371</td>
<td align="left">KXJ11891.1</td>
<td align="right">2.366390</td>
<td align="right">0.0000062</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.8662747</td>
<td align="right">3</td>
<td align="left">mitogen-activated protein kinase 1/3 [EC:2.7.11.24]</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Platelet activation</td>
<td align="left">Up</td>
</tr>
<tr class="odd">
<td align="left">ko04612</td>
<td align="left">K06696</td>
<td align="left">KXJ10301.1</td>
<td align="right">1.090056</td>
<td align="right">0.0013341</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.9076305</td>
<td align="right">4</td>
<td align="left">proteasome activator subunit 1 (PA28 alpha)</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Antigen processing and presentation</td>
<td align="left">Up</td>
</tr>
<tr class="even">
<td align="left">ko04620</td>
<td align="left">K04371</td>
<td align="left">KXJ11891.1</td>
<td align="right">2.366390</td>
<td align="right">0.0000062</td>
<td align="left">Vibrio Vibrio vs NoVibrio</td>
<td align="right">0.8662747</td>
<td align="right">3</td>
<td align="left">mitogen-activated protein kinase 1/3 [EC:2.7.11.24]</td>
<td align="left">Organismal Systems</td>
<td align="left">Immune system</td>
<td align="left">Toll-like receptor signaling pathway</td>
<td align="left">Up</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="references" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references">
<div id="ref-altschul_basic_1990">
<p>Altschul, S. F., W. Gish, W. Miller, E. W. Myers, and D. J. Lipman. 1990. “Basic Local Alignment Search Tool.” <em>Journal of Molecular Biology</em> 215 (3): 403–10. <a href="https://doi.org/10.1016/S0022-2836(05)80360-2">https://doi.org/10.1016/S0022-2836(05)80360-2</a>.</p>
</div>
<div id="ref-noauthor_database_2016">
<p>“Database Resources of the National Center for Biotechnology Information.” 2016. <em>Nucleic Acids Research</em> 44 (D1): D7–19. <a href="https://doi.org/10.1093/nar/gkv1290">https://doi.org/10.1093/nar/gkv1290</a>.</p>
</div>
<div id="ref-gentleman_annotate:_2018">
<p>Gentleman, R. 2018. <em>Annotate: Annotation for Microarrays</em>.</p>
</div>
<div id="ref-kanehisa_kegg:_2000">
<p>Kanehisa, M., and S. Goto. 2000. “KEGG: Kyoto Encyclopedia of Genes and Genomes.” <em>Nucleic Acids Research</em> 28 (1): 27–30.</p>
</div>
<div id="ref-kanehisa_kegg:_2017">
<p>Kanehisa, Minoru, Miho Furumichi, Mao Tanabe, Yoko Sato, and Kanae Morishima. 2017. “KEGG: New Perspectives on Genomes, Pathways, Diseases and Drugs.” <em>Nucleic Acids Research</em> 45 (D1): D353–D361. <a href="https://doi.org/10.1093/nar/gkw1092">https://doi.org/10.1093/nar/gkw1092</a>.</p>
</div>
<div id="ref-kanehisa_new_2019">
<p>Kanehisa, Minoru, Yoko Sato, Miho Furumichi, Kanae Morishima, and Mao Tanabe. 2019. “New Approach for Understanding Genome Variations in KEGG.” <em>Nucleic Acids Research</em> 47 (D1): D590–D595. <a href="https://doi.org/10.1093/nar/gky962">https://doi.org/10.1093/nar/gky962</a>.</p>
</div>
<div id="ref-love_moderated_2014">
<p>Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.” <em>Genome Biology</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.</p>
</div>
<div id="ref-tenenbaum_keggrest:_2018">
<p>Tenenbaum, Dan. 2018. <em>KEGGREST: Client-Side REST Access to KEGG</em>.</p>
</div>
<div id="ref-uniprotconsortium_uniprot:_2018">
<p>UniProt Consortium, The. 2018. “UniProt: The Universal Protein Knowledgebase.” <em>Nucleic Acids Research</em> 46 (5): 2699–9. <a href="https://doi.org/10.1093/nar/gky092">https://doi.org/10.1093/nar/gky092</a>.</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
